src/Data/ProgrammesDb/Util/SearchUtilitiesTrait.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\Util;
+
+trait SearchUtilitiesTrait
+{
+    protected function stripPunctuation(string $string): string
+    {
+        $string = preg_replace(
+            '/[\x{2000}-\x{206F}\x{2E00}-\x{2E7F}\'!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/u',
+            '',
+            $string
+        );
+        return $string ?? '';
+    }
+
+    protected function makeBooleanSearchQuery(string $keywords): ?string
+    {
+        $keywords = $this->stripPunctuation($keywords);
+        // Split into words
+        $words = preg_split('/\s+/', $keywords, 20, PREG_SPLIT_NO_EMPTY);
+        // Filter to words >= 2 chars
+        $validWords = array_filter($words, function ($word) {
```

Comment:
I learn something interesting about anonymous functions, you can declare them also as `static` so they don't get bound to the current class but this is a microptimization, doesn't really make a difference

Reply:
Interesting, didn't know you could do that. Can't see it making much difference here though.